### ISSUE TITLE
Ability to perform digest authentication

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [scalaj/scalaj-http#125]
Many tools like curl, Apache HttpComponents allow to do this.
I think only Basic auth is possible.

**Comment by hoffrocket** [comment]
Yes, currently on basic auth is supported. I'll take a stab at adding digest support.

**Comment by hoffrocket** [comment]
Hi, i just added this support in c060f856ea20018040e954e303c8d7c76209d3b4, but haven't tested in the real world. Can you clone the repo and test for your use case?
You can start a scala console with `./sbt console`
Example usage looks like: `scalaj.http.Http(url).digestAuth(user, pass).asString`

**Comment by DavidPerezIngeniero** [comment]
Thanks.

I've evaluated this expression with the right user/password combination and it returns `Unauthorized`:

```
Http("http://escucha03.fccma.com:8001/menu").digestAuth("user", "password").asString.body
```

**Comment by hoffrocket** [comment]
I Just pushed a potential fix. Could you pull the latest commits and try again?

**Comment by spydon** [comment]
I just tried this out and it works perfectly for me.
